scanner/indexer/indexer.go

```diff
@@ -2,43 +2,60 @@ package indexer
 
 import (
 	"context"
+	"errors"
+	"fmt"
 	"net/http"
+	"net/url"
 	"os"
+	"path"
+	"strings"
+	"time"
 
-	"github.com/pkg/errors"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/quay/claircore"
 	"github.com/quay/claircore/datastore/postgres"
 	"github.com/quay/claircore/libindex"
 	"github.com/quay/claircore/pkg/ctxlock"
+	"github.com/quay/zlog"
 	"github.com/stackrox/rox/pkg/utils"
 )
 
 // Indexer represents an image indexer.
-type Indexer struct {
+//
+//go:generate mockgen-wrapper
+type Indexer interface {
+	IndexContainerImage(context.Context, claircore.Digest, string, ...Option) (*claircore.IndexReport, error)
+	Close(context.Context) error
+}
+
+type indexerImpl struct {
 	indexer *libindex.Libindex
 }
 
 // NewIndexer creates a new indexer.
-func NewIndexer(ctx context.Context) (*Indexer, error) {
+func NewIndexer(ctx context.Context) (Indexer, error) {
 	// TODO: Update the conn string to something configurable.
 	pool, err := postgres.Connect(ctx, "postgresql:///postgres?host=/var/run/postgresql", "libindex")
 	if err != nil {
-		return nil, errors.Wrap(err, "connecting to postgres for indexer")
+		return nil, fmt.Errorf("connecting to postgres for indexer: %w", err)
```

Comment:
`pkg/errors` was obsoleted by [Go 1.13's extended error support](https://blog.golang.org/go1.13-errors). When Go 1.13 came out, the `"%w"` verb was added to `fmt.Errorf`, so now `errors.Wrapf` is obsolete. Reading that spec makes me feel we should not wrap the DB connection error and instead encapsulate it as part of the implementation. But since the indexer interface is internal, not consumed elsewhere, I would say it's OK to do it.

Reply:
Gotcha, and yea I agree with `pkg/errors` being obsolete - a few of my other changes reviewers requested the change from using `%w` to continue to use `pkg/errors` FWIW.